packages/neon/neon_dashboard/lib/src/pages/main.dart

```diff
@@ -156,6 +160,86 @@ class DashboardMainPage extends StatelessWidget {
               return const SizedBox.shrink();
             },
           ),
+          StreamBuilder(
+            stream: weatherStatusBloc.isSupported,
+            builder: (context, weatherStatusSupportedSnapshot) => ResultBuilder.behaviorSubject(
+              subject: weatherStatusBloc.forecasts,
+              builder: (context, forecastsResult) {
+                if (weatherStatusSupportedSnapshot.data ?? false) {
+                  Future<void> onWeatherStatusPressed() async {
+                    final location = await showDialog<String>(
+                      context: context,
+                      builder: (context) => DashboardSetWeatherLocationDialog(
+                        currentAddress: weatherStatusBloc.location.valueOrNull?.data?.address,
+                      ),
+                    );
+                    if (location != null) {
+                      weatherStatusBloc.setLocation(location);
+                    }
+                  }
+
+                  if (forecastsResult.hasData) {
+                    final weatherCode = forecastsResult.requireData.first.data.next1Hours.summary.symbolCode;
+                    final temperature = forecastsResult.requireData.first.data.instant.details.airTemperature;
+                    final description = switch (weatherCode.replaceAll(RegExp(r'_(day|night)$'), '')) {
+                      'clearsky' => DashboardLocalizations.of(context).weatherClearSky,
+                      'cloudy' => DashboardLocalizations.of(context).weatherCloudy,
+                      'fair' => DashboardLocalizations.of(context).weatherFairWeather,
+                      'partlycloudy' => DashboardLocalizations.of(context).weatherPartlyCloudy,
+                      'fog' => DashboardLocalizations.of(context).weatherFoggy,
+                      'rain' => DashboardLocalizations.of(context).weatherRainfall,
+                      'lightrain' => DashboardLocalizations.of(context).weatherLightRainfall,
+                      'heavyrain' => DashboardLocalizations.of(context).weatherHeavyRainfall,
+                      'rainshowers' => DashboardLocalizations.of(context).weatherRainfallShowers,
+                      'lightrainshowers' => DashboardLocalizations.of(context).weatherLightRainfallShowers,
+                      'heavyrainshowers' => DashboardLocalizations.of(context).weatherHeavyRainfallShowers,
+                      _ => throw UnimplementedError('Unknown weather code: $weatherCode'),
+                    };
```

Comment:
Please us an arb_file switch things to keep this cleaner.
I doubt we'll need to use these strings anywhere else in the project.